pkg/cleaner/finalizer_cleaner.go

```diff
@@ -0,0 +1,339 @@
+		_, err := client.ServicecatalogV1beta1().ServicePlans(toUpdate.Namespace).Update(toUpdate)
+		if err != nil {
+			return fmt.Errorf("failed to update ServicePlan %q: %s", toUpdate.Name, err)
+		}
+		err = wait.Poll(finalizerCheckPerdiodTime, finalizerCheckTimeout, func() (done bool, err error) {
+			klog.V(4).Info("waiting for the finalizer to be removed")
+			cr, err := client.ServicecatalogV1beta1().ServicePlans(toUpdate.Namespace).Get(toUpdate.Name, v1.GetOptions{})
+			return checkFinalizerIsRemoved(cr, err)
+		})
+		if err != nil {
+			return fmt.Errorf("failed while waiting for finalizers will be removed: %s", err)
+		}
+	}
+
+	return nil
+}
+
+func removeFinalizerFromServiceInstance(client sc.Interface) error {
+	list, err := client.ServicecatalogV1beta1().ServiceInstances(v1.NamespaceAll).List(v1.ListOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("failed to list ServiceInstance: %s", err)
+	}
+
+	for _, instance := range list.Items {
+		finalizersList := removeServiceCatalogFinalizer(instance.Finalizers)
+		toUpdate := instance.DeepCopy()
+		toUpdate.Finalizers = finalizersList
+		_, err := client.ServicecatalogV1beta1().ServiceInstances(toUpdate.Namespace).Update(toUpdate)
+		if err != nil {
+			return fmt.Errorf("failed to update ServiceInstance %q: %s", toUpdate.Name, err)
+		}
+		err = wait.Poll(finalizerCheckPerdiodTime, finalizerCheckTimeout, func() (done bool, err error) {
+			klog.V(4).Info("waiting for the finalizer to be removed")
+			cr, err := client.ServicecatalogV1beta1().ServiceInstances(toUpdate.Namespace).Get(toUpdate.Name, v1.GetOptions{})
+			return checkFinalizerIsRemoved(cr, err)
+		})
+		if err != nil {
+			return fmt.Errorf("failed while waiting for finalizers will be removed: %s", err)
+		}
+	}
+
+	return nil
+}
+
+func removeFinalizerFromServiceBinding(client sc.Interface) error {
+	list, err := client.ServicecatalogV1beta1().ServiceBindings(v1.NamespaceAll).List(v1.ListOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("failed to list ServiceBinding: %s", err)
+	}
+
+	for _, binding := range list.Items {
+		finalizersList := removeServiceCatalogFinalizer(binding.Finalizers)
+		toUpdate := binding.DeepCopy()
+		toUpdate.Finalizers = finalizersList
+		_, err := client.ServicecatalogV1beta1().ServiceBindings(toUpdate.Namespace).Update(toUpdate)
+		if err != nil {
+			return fmt.Errorf("failed to update ServiceBinding %q: %s", toUpdate.Name, err)
+		}
+		err = wait.Poll(finalizerCheckPerdiodTime, finalizerCheckTimeout, func() (done bool, err error) {
+			klog.V(4).Info("waiting for the finalizer to be removed")
+			cr, err := client.ServicecatalogV1beta1().ServiceBindings(toUpdate.Namespace).Get(toUpdate.Name, v1.GetOptions{})
+			return checkFinalizerIsRemoved(cr, err)
+		})
+		if err != nil {
+			return fmt.Errorf("failed while waiting for finalizers will be removed: %s", err)
+		}
+	}
+
+	return nil
+}
+
+type CRWithFinalizer interface {
```

Comment:
```suggestion
type FinalizerGetter interface {
```

in go the interface name are generally named like, statusGetter, ReadCloser, xxxUpdateDeleter etc